network-config-analyzer/K8sServiceYamlParser.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+import Peer
+import yaml
+from PeerContainer import PeerContainer
+from K8sService import K8sService
+from CmdlineRunner import CmdlineRunner
+from GenericTreeScanner import TreeScannerFactory
+from K8sYamlParser import K8sYamlParser
+
+
+class K8sServiceYamlParser(K8sYamlParser):
+    """
+    A parser for k8s service resources
+    """
+
+    def __init__(self, service_file_name=''):
+        """
+        :param str service_file_name: The name of the yaml file containing K8s service resources
+        """
+        K8sYamlParser.__init__(self, service_file_name)
+
+    def parse_service(self, srv_object, peer_container):
+        """
+        Parses a service resource object and creates a K8sService object
+        :param dict srv_object: the service object to parse
+        :param PeerContainer peer_container: the peer container in which pods of the service are located
+        :return: K8sService object or None
+        """
+        if srv_object.get('kind') != 'Service':
+            return None  # Not a Service object
+        api_version = srv_object.get('apiVersion')
+        if api_version != 'v1':
+            return None  # apiVersion is not properly set
+        metadata = srv_object.get('metadata')
+        if not metadata:
+            return None
+        srv_name = metadata.get('name')
+        if not srv_name:
+            return None
+        self.check_dns_subdomain_name(srv_name, metadata)
+        service_spec = srv_object.get('spec')
+        if not service_spec:
+            self.warning(f'Spec is missing or null in Service {srv_name}. Ignoring the service')
+            return None
+        service = K8sService(srv_name)
+        service_type = service_spec.get('type', 'ClusterIP')
+        if service_type == 'ExternalName':
+            service.set_type(K8sService.ServiceType.ExternalName)
+        elif service_type == 'NodePort':
+            service.set_type(K8sService.ServiceType.NodePort)
+        elif service_type == 'LoadBalancer':
+            service.set_type(K8sService.ServiceType.LoadBalancer)
+        else:
+            service.set_type(K8sService.ServiceType.ClusterIP) # the default type
+
+        selector = service_spec.get('selector')
+        if selector:
+            for key, val in selector.items():
+                self.check_label_key_syntax(key, selector)
+                self.check_label_value_syntax(val, key, selector)
+                service.add_selector(key, val)
+                service.target_pods |= peer_container.get_peers_with_label(key, [val])
+
+        ports = service_spec.get('ports')
+        if ports is not None:
+            for port in ports:
+                port_id = port.get('port')
+                if not port_id:
+                    continue
+                self.check_port_syntax(port_id, False, 'Service port')
```

Comment:
can we remove port syntax check to simplify, and assume it's correct?

Reply:
Removed port syntax check